agent/exec/container/adapter.go

```diff
@@ -61,15 +62,28 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 
 	dec := json.NewDecoder(rc)
 	m := map[string]interface{}{}
+	spamLimiter := rate.NewLimiter(rate.Every(1000*time.Millisecond), 1)
+
 	for {
 		if err := dec.Decode(&m); err != nil {
 			if err == io.EOF {
 				break
 			}
 			return err
 		}
-		// TODO(stevvooe): Report this status somewhere.
-		logrus.Debugln("pull progress", m)
+		if spamLimiter.Allow() {
+			if progress, ok := m["progressDetail"].(map[string]interface{}); ok {
+				// only post pull progress to log every couple of seconds
+				log.G(ctx).WithFields(logrus.Fields{
+					"id":      m["id"],
+					"status":  m["status"],
+					"current": progress["current"],
+					"total":   progress["total"],
+				}).Debug("pull in progress")
```

Comment:
A few things:

- lose the id; this should be covered by having `task.id` on the contextual logger. I am not sure we propagate this, since we actually fork the context to change the lifecycle.
- When the value of `current` and `total` are `nil`, don't report them. I am not sure where this is coming from, but it is likely an `interface{}` value which is causing this.
- Include the name of the image. This is fairly helpful in practice.